Gems/ROS2/Code/Source/RobotImporter/URDF/SensorsMaker.cpp

```diff
@@ -40,7 +40,7 @@ namespace ROS2
         for (size_t si = 0; si < link->SensorCount(); ++si)
         {
             const auto* sensor = link->SensorByIndex(si);
-            const bool success = AddSensor(entityId, sensor);
+            [[maybe_unused]] const bool success = AddSensor(entityId, sensor);
```

Comment:
Another option for this is to use [AZ_VerifyWarning](https://github.com/o3de/o3de/blob/c4525e658059abd2fdf7a520e8120f66a917c306/Code/Framework/AzCore/AzCore/Debug/Trace.h#L406)

```
const bool success = AddSensor(entityId, sensor);
AZ_VerifyWarning("SensorMaker", success, "Cannot find a sensor hook for sensor %d", sensor->Type());
```